plugin/trino-hive/pom.xml

```diff
@@ -251,6 +256,12 @@
         </dependency>
 
         <!-- used by tests but also needed transitively -->
+        <dependency>
+            <groupId>io.trino</groupId>
+            <artifactId>trino-parser</artifactId>
```

Comment:
the trino-parser was supposed to be shaded, per @shenodaguirguis's https://github.com/trinodb/trino/pull/10226#discussion_r768046300

Reply:
Please see https://github.com/linkedin/coral/issues/205#issuecomment-1017829694 and https://github.com/linkedin/coral/pull/222 

`trino-parser` was transformed into a compile-time dependency in the meantime for the coral project

Reply:
then we're back at square one, per https://github.com/trinodb/trino/pull/10226#discussion_r764858841

Reply:
https://mvnrepository.com/artifact/com.linkedin.coral/coral-trino-parser/2.0.53 

There is a shaded version of the `coral-trino-parser` library.
However, this is unfortunately empty - containing no files at all.

Reply:
cc: @wmoustafa

Reply:
Discussed this offline with @findinpath. So exclusion was on the table before, but I remember we agreed not to go that route (cannot find the comment now though). Shading should fix the issue, and we integrated the shaded version internally. @ljfgem since you did the integration internally, could you take a look at this integration? Also, the shaded library should be `coral-trino-parser-all`.

Reply:
For this upgrade, given `trino-parser` used in Coral-Trino has been shaded, I don't think we need to change any `trino-parser` configs in this file, `plugin/trino-iceberg/pom.xml` or `pom.xml`

Reply:
@findinpath can you please verify the shading now works?

Reply:
`coral-trino-parser` dependency is empty. 
I have also reported this issue here: https://github.com/linkedin/coral/issues/233

On the other hand, `trino-parser` is a `compileOnly` (aka `provided` in `maven) dependency as it can be seen here: https://github.com/linkedin/coral/commit/5ea65e82caef38ad109a8f91745fbf058dd5dfed

So I'd say there (currently) no need to worry about shading.

Reply:
provided means it's on the user of library to provide the dependency,
doesn't mean the dependency is not needed
and doesn't mean the dependency can have breaking API changes without breaking the library itself
(see also https://github.com/trinodb/trino/pull/10226#discussion_r764846874)

i think we're making circles. the current state doesn't seem any better than were we started.

because
1. we believe trino-parser is not needed at all
2. we also believe the coral-trino module will be fixed eventually, probably sharing trino-parser correctly

i am fine with merging an imperfect solution for now.
i think the best what we can do is **exclude** trino-parser dependency from the coral dependency:

- trino-parser remains test-only dependency of trino-hive, and is not part of the plugin
- if our assumptions are wrong, and our use of coral indeed requires (or starts to require) trino-parser, we will know about this in a very clear `NoClassDefFound` message, instead of some less predictable failure (including potential silent correctness issues)

Reply:
I still not following why we cannot just remove the dependency on `trino-parser` (or keep it to `test` as it was the case before).
`coral-trino-parser-2.0.53.jar` is empty -- we can look into not publishing this but this should be irrelevant.
`coral-trino-parser-2.0.53-all.jar` has the shaded jar and is not empty.
Coral internally uses `coral-trino-parser-2.0.53-all.jar`, so this should not conflict with anything on Trino DB side (hence this file should not be changed).

If you revert this change, and still run to the issue on local builds, most likely you need to clean up your local maven caches. We ran to this issue and cleaning local maven caches helps.

Reply:
From `mvn dependency:tree`:
```
[INFO] +- com.linkedin.coral:coral-trino:jar:2.0.53:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.8.1:compile
[INFO] |  +- com.linkedin.coral:coral-trino-parser:jar:2.0.53:compile
```

It very much looks like `coral-trino.jar` depends on `coral-trino-parser-2.0.53.jar`. Also this is what was pulled to my `.m2/repository` when I built trino from this PR. And the jar is empty.

Reply:
@ljfgem do you have the corresponding tree from our internal integration?

Reply:
@losipiuk @findinpath @findepi @wmoustafa 
Yes, we have the similar corresponding tree from our internal integration.
But I tried 2 builds:
1. Build without any config changes for trino-parser, just upgrade Coral to v2 and modify corresponding codes
2. Build with the trino-parser config changes (just applied this PR)
and both builds succeeded.
I am wondering which building issue we are facing, could you provide the stacktrace?